docs/data-flow.md

```diff
@@ -0,0 +1,103 @@
+- [ErpProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/erp/models/requests/ProducerRequest.ts)
+
+### `user-service` key
+
+`serverless-user-service` payloads are meant to be completed by `supporter-event-service`. In particular, `supporter-event-service` will loop through `identities` and hash `identities.content` and add it as `identities.uuid` key if it is not already defined, effectively transforming a `PartialSignupRequest` (accepted by `supporter-event-service`) into a `SignupRequest` (accepted by `serverless-user-service`).
+
+See:
+- [Identity](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/Identity.ts)
+- [PartialSignupRequest and SignupRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/requests/SignupRequest.ts)
+
+### `erp-service` key
+
+`serverless-erp-service` payloads are meant to be completed by `supporter-event-service`. You can see [check out the specific implementations on `data-model`](https://github.com/comicrelief/data-models/tree/master/src/schemas/erp/models/actions). They consists of simple object contains an array of Erp Actions.
+
+```javascript
+{
+  "actions": ErpAction[]
+}
+```
+
+See:
+- [ErpProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/erp/models/requests/ProducerRequest.ts)
+- [ErpProducerRequest examples (mocks)](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/mocks/requests/ProducerRequest.mock.ts)
+
+Each ErpAction should validate against the following:
+
+```javascript
+{
+  "type": ErpPayloadTypes,
+  "data": {
+    // Upstream service specific data
+    specificData1: any,
+    specificData2: any,
+
+    // The following key is likely to be required
+    // for supporter-specific payloads
+    // such as donations, prize tickets, etc.
+    // See the `Supporter UUID` section for details
+    supporterUuid: String,
+
+    // If your service doesn't have access
+    // to supporter specific information
+    // the payload will need to specify
+    // the `identityHash` key so that `supporter-event-service`
+    // will be able to complete the payload
+    identityHash: String
+  }
+}
+```
+
+See:
+- [ErpPayloadTypes](https://github.com/comicrelief/data-models/blob/master/src/schemas/erp/models/common.ts#ErpPayloadTypes)
+- [Erp Actions List](https://github.com/comicrelief/data-models/tree/master/src/schemas/erp/models/actions)
+
+## Supporter UUID
+
+Your ERP payloads may be related to a specific supporter. This is true for transactions, prize tickets, Shopify orders, etc. The `data.supporterUuid` key will be used by ERP Next to link the entity with a Supporter entity. However `supporterUuid` cannot be guaranteed to upstream services as it will be stored on `serverless-user-service`. This is why `supporter-event-service` can help your upstream service complete its payload.
```

Comment:
It is explained in scenario 1 below, and it means you don't have to send `supporterUuid` in the erp-service payload, if you are already sending a user-service payload.
Maybe the wording is not clear enough!

Reply:
It is that idea. We are going to send incomplete payloads to `supporter-event-service`, which will be updated with additional information available there. Not sure `decorate` works (at least, not for me, since I come from a Python background!)

Reply:
Agreed it's not quite right. Update? Append? Add to? Extend?

Reply:
they all lack the idea that if you send that payload to downstream services, they will 400 you

Reply:
Maybe just spell it out a bit more clearly the first time you say 'complete'?  I genuinely didn't get the sense in which you meant it but then again maybe I was just being dim

Reply:
What do you think of this change?
af626a2dbf703657dbd5b5739f069436d3fae4a1

Reply:
That's great, cheers!

Reply:
The explanation of terms is helpful, thanks. I think it would be useful to also give an example, basically to answer the question *why would I be sending incomplete records?* The answer as I understand it being: *if you need to link the record to a user, you need the user ID, which you don't have yet but which supporter-event-service is going to find/create for you*.

Also, how about changing "a few keys" -> "the missing required keys"?